actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2468,10 +2468,22 @@ def submit(value = nil, options = {})
       #   #      <strong>Ask me!</strong>
       #   #    </button>
       #
+      #   button do |text|
+      #     content_tag(:strong, text)
+      #   end
+      #   # => <button name='button' type='submit'>
+      #   #      <strong>Create post</strong>
+      #   #    </button>
+      #
       def button(value = nil, options = {}, &block)
         value, options = nil, value if value.is_a?(Hash)
         value ||= submit_default_value
-        @template.button_tag(value, options, &block)
+
+        if block_given?
+          value = @template.capture { yield(value) }
```

Comment:
@rafaelfranca I've pushed up https://github.com/rails/rails/pull/38550/commits/b1c91250dae445daadc41e4d7e1ed1a797b44099 to replace the `proc { yield(value) }` with a call to [`capture`](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/CaptureHelper.html#method-i-capture), is that an improvement?

Reply:
@rafaelfranca is this worth considering for 6.1?